tests/unit/s2n_tls_prf_test.c

```diff
@@ -136,5 +137,90 @@ int main(int argc, char **argv)
+     * Here we test that the retrieved digest is the same as the digest after the Client Key Exchange
+     * message is added to the transcript hash.
+     *
+     *= https://tools.ietf.org/rfc/rfc7627#section-3
+     *= type=test
+     *# When a full TLS handshake takes place, we define
+     *#
+     *#       session_hash = Hash(handshake_messages)
+     *#
+     *# where "handshake_messages" refers to all handshake messages sent or
+     *# received, starting at the ClientHello up to and including the
+     *# ClientKeyExchange message, including the type and length fields of
+     *# the handshake messages.
+     */
+    {
+        struct s2n_cert_chain_and_key *tls12_chain_and_key = NULL;
+        EXPECT_SUCCESS(s2n_test_cert_chain_and_key_new(&tls12_chain_and_key, S2N_DEFAULT_TEST_CERT_CHAIN,
+                S2N_DEFAULT_TEST_PRIVATE_KEY));
+
+        struct s2n_config *config = s2n_config_new();
+        EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(config, tls12_chain_and_key));
+        EXPECT_SUCCESS(s2n_config_set_unsafe_for_testing(config));
+        EXPECT_SUCCESS(s2n_config_set_cipher_preferences(config, "20170210"));
+
+        struct s2n_connection *client_conn = s2n_connection_new(S2N_CLIENT);
+        EXPECT_NOT_NULL(client_conn);
+
+        struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER);
+        EXPECT_NOT_NULL(server_conn);
+
+        EXPECT_SUCCESS(s2n_connection_set_config(client_conn, config));
+        EXPECT_SUCCESS(s2n_connection_set_config(server_conn, config));
+
+        DEFER_CLEANUP(struct s2n_stuffer client_to_server = { 0 }, s2n_stuffer_free);
+        DEFER_CLEANUP(struct s2n_stuffer server_to_client = { 0 }, s2n_stuffer_free);
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&client_to_server, 0));
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&server_to_client, 0));
+        EXPECT_SUCCESS(s2n_connection_set_io_stuffers(&server_to_client, &client_to_server, client_conn));
+        EXPECT_SUCCESS(s2n_connection_set_io_stuffers(&client_to_server, &server_to_client, server_conn));
+
+        EXPECT_OK(s2n_negotiate_test_server_and_client_until_message(server_conn, client_conn, CLIENT_KEY));
+
+        /* Client writes Client Key Exchange message */
+        EXPECT_SUCCESS(s2n_handshake_write_io(client_conn));
+
+        uint8_t data[S2N_MAX_DIGEST_LEN] = { 0 };
+        struct s2n_blob digest_for_ems = { 0 };
+        EXPECT_SUCCESS(s2n_blob_init(&digest_for_ems, data, sizeof(data)));
+
+        /* Get the Client Key transcript */
+        EXPECT_SUCCESS(s2n_stuffer_skip_read(&client_to_server, S2N_TLS_RECORD_HEADER_LENGTH));
+        uint8_t client_key_message_length = s2n_stuffer_data_available(&client_to_server);
+        struct s2n_blob message = { 0 };
+        message.data = &client_to_server.blob.data[client_to_server.read_cursor];
+        message.size = client_key_message_length;
```

Comment:
Probably a good idea to call s2n_blob_init instead of setting these manually, and call s2n_stuffer_raw_read instead of accessing the data array directly.